From e31ac6f4e914d210c14c21fc0bd149e39178ca39 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Tue, 13 Jan 2015 22:19:53 +0800
Subject: [PATCH 1359/1543] MLK-10089 arm: imx: Add 198MHz OPP for i.MX6SX

commit 1ec6ce89a5fb6645ffc160f9ecbb8c86ba910347 from
git://git.freescale.com/imx/linux-2.6-imx.git

Update the i.MX6SX operating points to comply with the latest
datasheet. Latest i.MX6SX datasheet of Rev.F, 1/2015 adds the
198MHz setpoint. For the RevB board, the VDD_ARM and ADD_SOC
are connected together, so the voltage for 198MHz needs to be
set to 1.175V. for the general setting, add a 25mV margin to
cover the board IR drop.

this patch is copy from commit: 17dae6e92 on branch imx_3.10.y
as too many conflicts to resolve.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sdb-ldo.dts  |    2 ++
 arch/arm/boot/dts/imx6sx-sdb-reva.dts |    2 ++
 arch/arm/boot/dts/imx6sx-sdb.dts      |    2 ++
 arch/arm/boot/dts/imx6sx.dtsi         |    2 ++
 4 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-ldo.dts b/arch/arm/boot/dts/imx6sx-sdb-ldo.dts
index 6128ec2..d9b3acc 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-ldo.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-ldo.dts
@@ -14,12 +14,14 @@
 		996000  1250000
 		792000  1175000
 		396000  1075000
+		198000	975000
 		>;
 	fsl,soc-operating-points = <
 		/* ARM kHz      SOC uV */
 		996000	1175000
 		792000	1175000
 		396000	1175000
+		198000	1175000
 	>;
 	arm-supply = <&reg_arm>;
 	soc-supply = <&reg_soc>;
diff --git a/arch/arm/boot/dts/imx6sx-sdb-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
index 1610260..d978373 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-reva.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
@@ -20,12 +20,14 @@
 		996000  1250000
 		792000  1175000
 		396000  1075000
+		198000	975000
 		>;
 	fsl,soc-operating-points = <
 		/* ARM kHz      SOC uV */
 		996000	1175000
 		792000	1175000
 		396000	1175000
+		198000	1175000
 	>;
 	arm-supply = <&sw1a_reg>;
 	soc-supply = <&sw1c_reg>;
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index f5cc42d..5d977fe 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -217,12 +217,14 @@
 		996000  1250000
 		792000  1175000
 		396000  1175000
+		198000	1175000
 		>;
 	fsl,soc-operating-points = <
 		/* ARM kHz      SOC uV */
 		996000	1250000
 		792000	1175000
 		396000	1175000
+		198000	1175000
 	>;
 	arm-supply = <&sw1a_reg>;
 	soc-supply = <&sw1a_reg>;
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index a98ed98..79b9e5ec 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -62,12 +62,14 @@
 				996000  1250000
 				792000  1175000
 				396000  1075000
+				198000	975000
 			>;
 			fsl,soc-operating-points = <
 				/* ARM kHz  SOC uV */
 				996000      1175000
 				792000      1175000
 				396000      1175000
+				198000	    1175000
 			>;
 			clock-latency = <61036>; /* two CLK32 periods */
 			clocks = <&clks IMX6SX_CLK_ARM>,
-- 
1.7.5.4

